codenvy-api-factory/src/main/java/com/codenvy/api/factory/FactoryService.java

```diff
@@ -350,6 +350,78 @@ public void removeFactory(@ApiParam(value = "Factory ID", required = true)
         factoryStore.removeFactory(id);
     }
 
+
+    /**
+     * Updates factory with a new factory content
+     *
+     * @param id
+     *         id of factory
+     * @param newFactory
+     *         the new data for the factory
+     * @throws NotFoundException
+     *         when factory with given id doesn't exist
+     */
+    @ApiOperation(value = "Updates factory information by its ID",
+            notes = "Updates factory based on the Factory ID which is passed in a path parameter")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 403, message = "User not authorized to call this operation"),
+            @ApiResponse(code = 404, message = "Factory not found"),
+            @ApiResponse(code = 500, message = "Internal Server Error")})
+    @POST
+    @Path("/{id}")
+    @RolesAllowed("user")
+    @Consumes({MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_JSON})
+    public Factory updateFactory(@ApiParam(value = "Factory ID", required = true)
+                                 @PathParam("id") String id,
+                                 Factory newFactory) throws ForbiddenException, NotFoundException, ApiException {
+
+        // Check we've a user
+        final User user = EnvironmentContext.getCurrent().getUser();
+        if (user == null) {
+            // well this shouldn't happen if only user is authorized to call the method
+            throw new ForbiddenException("No authenticated user");
+        }
+
+        // Do we have a factory for this id ?
+        Factory existingFactory = factoryStore.getFactory(id);
+        if (existingFactory == null) {
+            throw new NotFoundException("Factory with id " + id + " does not exist.");
+        }
+
+        // Gets the User id
+        String userId = user.getId();
+
+        // Validate the factory against the current user
+        factoryEditValidator.validate(existingFactory, userId);
+
+        // Check author is set and copy created date from old factory
+        Author newAuthor = newFactory.getCreator();
+        if (newAuthor == null) {
+            newAuthor = DtoFactory.getInstance().createDto(Author.class);
+            newFactory.setCreator(newAuthor);
+        }
```

Comment:
as we can't change author, do we really care of what is submitted by the new factory ? as we can add back the author userid of the previous factory